### DISCUSSION TITLE
Best way to set global font?

### DISCUSSION THREAD

**Discussion by joeyscarim** [nativewind/nativewind#185] [General]
I have a font setup, and added to my config:

```
  theme: {
    extend: {
      fontFamily: {
        poppins: ['Poppins'],
      },
    },
  },
```

This works, but I have to add `font-poppins` to every <Text> component throughout my app.

To set it globally, I tried to override the "sans" font (which looking back I realize doesnt make sense at all for ios and android:

```
  theme: {
    extend: {
      fontFamily: {
        poppins: ['Poppins'],
      },
    },
  },
```

I then tried to make a `FontText` component that applies the font, and passes the other props. However, this doesnt work with the `className` prop (maybe because of how the babel plugin works?

```
import React from 'react';
import {Text} from 'react-native';

export const FontText = (props: any) => (
  <Text
    {...props}
    style={{...props.style, fontFamily: 'Poppins'}}>
    {props.children}
  </Text>
);
```

So I then I renamed the className prop on my <FontText> components to `textClassName`, and set that to className in my FontText component:

```
import React from 'react';
import {Text} from 'react-native';

export const FontText = (props: any) => (
  <Text
    className={props.textClassName}
    {...props}
    style={{...props.style, fontFamily: 'Poppins'}}>
    {props.children}
  </Text>
);
```

Which works, but feels very unergonomic. Am I missing a simpler way to accomplish this? 

Thanks for any tips!

**Comment by marklawlor** [comment]
You can use the `styled()` api to create a styled Text component with default classNames

https://www.nativewind.dev/api/styled#default-styles

```tsx
const MyText = styled(Text, "font-poppins")

<MyText>Hello World.</MyText>
```

> However, this doesnt work with the className prop (maybe because of how the babel plugin works?

Please read the docs on custom components https://www.nativewind.dev/guides/custom-components

**Comment by joeyscarim** [comment]
Thank you! `styled()` api is exactly what I was looking for
